src/config.rs

```diff
@@ -152,6 +165,46 @@ impl Config {
         self.toml.as_ref().map_or(Ok(None), |t| Ok(t.image(target)))
     }
 
+    pub fn dockerfile(&self, target: &Target) -> Result<Option<String>> {
+        let env_value = self.env.dockerfile(target);
+        if let Some(env_value) = env_value {
+            return Ok(Some(env_value));
+        }
+        self.toml.as_ref().map_or(Ok(None), |t| {
+            Ok(t.dockerfile(target).map(ToOwned::to_owned))
+        })
+    }
+
+    pub fn dockerfile_context(&self, target: &Target) -> Result<Option<String>> {
+        let env_value = self.env.dockerfile_context(target);
+        if let Some(env_value) = env_value {
+            return Ok(Some(env_value));
+        }
+        self.toml.as_ref().map_or(Ok(None), |t| {
+            Ok(t.dockerfile_context(target).map(ToOwned::to_owned))
+        })
+    }
+
+    pub fn dockerfile_build_args(
+        &self,
+        target: &Target,
+    ) -> Result<Option<HashMap<String, String>>> {
+        // This value does not support env variables
```

Comment:
would it make sense to allow this? How?

Reply:
I doubt it, since the build args could have characters that we'd expect in any delimiter we choose.